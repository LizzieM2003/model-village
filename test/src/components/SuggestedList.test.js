import React from 'react';

import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { SuggestedList } from './SuggestedList';
import suggestedData from '../data/suggested_influencers.json';

configure({ adapter: new Adapter() });

describe('SuggestedList', () => {
  const mockFetchInfluencer = jest.fn();
  const props = {
    suggested: suggestedData
  };
  const suggestedList = shallow(<SuggestedList {...props} />);

  it('renders correctly', () => {
    expect(suggestedList).toMatchSnapshot();
  });

  it('has one SuggestedItem component for each suggested influencer', () => {
    expect(suggestedList.find('.SuggestedList-items').children().length).toBe(3);
  });
});
