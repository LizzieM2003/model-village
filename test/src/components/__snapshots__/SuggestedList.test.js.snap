// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestedList renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SuggestedList
    suggested={
        Array [
            Object {
              "influencer_full_name": "Hannah Ross",
              "influencer_id": 123,
              "influencer_instagram_profile_image": "https://randomuser.me/api/portraits/women/62.jpg",
              "influencer_instagram_username": "hross64346",
            },
            Object {
              "influencer_full_name": "Emily White",
              "influencer_id": 234,
              "influencer_instagram_profile_image": "https://randomuser.me/api/portraits/women/14.jpg",
              "influencer_instagram_username": "emwhite",
            },
            Object {
              "influencer_full_name": "Olivia Wagner",
              "influencer_id": 345,
              "influencer_instagram_profile_image": "https://randomuser.me/api/portraits/women/29.jpg",
              "influencer_instagram_username": "olivia_wagner_official",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Suggested Influencers
</h2>,
        <div
          className="SuggestedList-items"
>
          <Connect(Component)
                    influencer_full_name="Hannah Ross"
                    influencer_id={123}
                    influencer_instagram_profile_image="https://randomuser.me/api/portraits/women/62.jpg"
                    influencer_instagram_username="hross64346"
          />
          <Connect(Component)
                    influencer_full_name="Emily White"
                    influencer_id={234}
                    influencer_instagram_profile_image="https://randomuser.me/api/portraits/women/14.jpg"
                    influencer_instagram_username="emwhite"
          />
          <Connect(Component)
                    influencer_full_name="Olivia Wagner"
                    influencer_id={345}
                    influencer_instagram_profile_image="https://randomuser.me/api/portraits/women/29.jpg"
                    influencer_instagram_username="olivia_wagner_official"
          />
</div>,
      ],
      "className": "SuggestedList",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Suggested Influencers",
        },
        "ref": null,
        "rendered": "Suggested Influencers",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(Component)
              influencer_full_name="Hannah Ross"
              influencer_id={123}
              influencer_instagram_profile_image="https://randomuser.me/api/portraits/women/62.jpg"
              influencer_instagram_username="hross64346"
/>,
            <Connect(Component)
              influencer_full_name="Emily White"
              influencer_id={234}
              influencer_instagram_profile_image="https://randomuser.me/api/portraits/women/14.jpg"
              influencer_instagram_username="emwhite"
/>,
            <Connect(Component)
              influencer_full_name="Olivia Wagner"
              influencer_id={345}
              influencer_instagram_profile_image="https://randomuser.me/api/portraits/women/29.jpg"
              influencer_instagram_username="olivia_wagner_official"
/>,
          ],
          "className": "SuggestedList-items",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "123",
            "nodeType": "class",
            "props": Object {
              "influencer_full_name": "Hannah Ross",
              "influencer_id": 123,
              "influencer_instagram_profile_image": "https://randomuser.me/api/portraits/women/62.jpg",
              "influencer_instagram_username": "hross64346",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "234",
            "nodeType": "class",
            "props": Object {
              "influencer_full_name": "Emily White",
              "influencer_id": 234,
              "influencer_instagram_profile_image": "https://randomuser.me/api/portraits/women/14.jpg",
              "influencer_instagram_username": "emwhite",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "345",
            "nodeType": "class",
            "props": Object {
              "influencer_full_name": "Olivia Wagner",
              "influencer_id": 345,
              "influencer_instagram_profile_image": "https://randomuser.me/api/portraits/women/29.jpg",
              "influencer_instagram_username": "olivia_wagner_official",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Suggested Influencers
</h2>,
          <div
            className="SuggestedList-items"
>
            <Connect(Component)
                        influencer_full_name="Hannah Ross"
                        influencer_id={123}
                        influencer_instagram_profile_image="https://randomuser.me/api/portraits/women/62.jpg"
                        influencer_instagram_username="hross64346"
            />
            <Connect(Component)
                        influencer_full_name="Emily White"
                        influencer_id={234}
                        influencer_instagram_profile_image="https://randomuser.me/api/portraits/women/14.jpg"
                        influencer_instagram_username="emwhite"
            />
            <Connect(Component)
                        influencer_full_name="Olivia Wagner"
                        influencer_id={345}
                        influencer_instagram_profile_image="https://randomuser.me/api/portraits/women/29.jpg"
                        influencer_instagram_username="olivia_wagner_official"
            />
</div>,
        ],
        "className": "SuggestedList",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Suggested Influencers",
          },
          "ref": null,
          "rendered": "Suggested Influencers",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(Component)
                influencer_full_name="Hannah Ross"
                influencer_id={123}
                influencer_instagram_profile_image="https://randomuser.me/api/portraits/women/62.jpg"
                influencer_instagram_username="hross64346"
/>,
              <Connect(Component)
                influencer_full_name="Emily White"
                influencer_id={234}
                influencer_instagram_profile_image="https://randomuser.me/api/portraits/women/14.jpg"
                influencer_instagram_username="emwhite"
/>,
              <Connect(Component)
                influencer_full_name="Olivia Wagner"
                influencer_id={345}
                influencer_instagram_profile_image="https://randomuser.me/api/portraits/women/29.jpg"
                influencer_instagram_username="olivia_wagner_official"
/>,
            ],
            "className": "SuggestedList-items",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "123",
              "nodeType": "class",
              "props": Object {
                "influencer_full_name": "Hannah Ross",
                "influencer_id": 123,
                "influencer_instagram_profile_image": "https://randomuser.me/api/portraits/women/62.jpg",
                "influencer_instagram_username": "hross64346",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "234",
              "nodeType": "class",
              "props": Object {
                "influencer_full_name": "Emily White",
                "influencer_id": 234,
                "influencer_instagram_profile_image": "https://randomuser.me/api/portraits/women/14.jpg",
                "influencer_instagram_username": "emwhite",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "345",
              "nodeType": "class",
              "props": Object {
                "influencer_full_name": "Olivia Wagner",
                "influencer_id": 345,
                "influencer_instagram_profile_image": "https://randomuser.me/api/portraits/women/29.jpg",
                "influencer_instagram_username": "olivia_wagner_official",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
